pkg/pillar/cmd/vaultmgr/vaultmgr.go

```diff
@@ -722,6 +785,9 @@ func Run(ps *pubsub.PubSub, loggerArg *logrus.Logger, logArg *base.LogObject) in
 				subVaultKeyFromController.ProcessChange(change)
 			case <-stillRunning.C:
 				ps.StillRunning(agentName, warningTime, errorTime)
+			case <-ctx.ucChan:
+				//Publish current status of vault
+				publishVaultStatus(&ctx)
```

Comment:
This seems error-prone. If you some code calls publishVaultStatus earlier, the conversion state will be off.
Thus I think it makes sense to make that explicit by adding a ConversionComplete bool to vaultStatus and have WaitForVault wait for that to become true.

Reply:
Updated the changes accordingly, please check